### PR TITLE
Bugfixes 2.7 swift3

### PR DESCRIPTION
**Pull request by lastMove**
Fix MaskType .none in the playground part of the app. 
Fix Gradient & PredefinedGradient
Fix TransitionType problems (needs some testing)

### CODE REVIEW

File: IBAnimatable/CardsAnimator.swift [hunk 1]
```diff
@@ -35,7 +35,7 @@ public class CardsAnimator: NSObject, AnimatedTransitioning {
 
 extension CardsAnimator: UIViewControllerAnimatedTransitioning {
   public func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
-    return transitionDuration(using: transitionContext)
+    return retrieveTransitionDuration(transitionContext: transitionContext)
```
Comment (JakeLin) [inline IBAnimatable/CardsAnimator.swift:38]: My bad😁

File: IBAnimatable/MaskDesignable.swift [hunk 2]
```diff
@@ -50,7 +50,7 @@ public extension MaskDesignable where Self: UIView {
     case .triangle:
       maskTriangle()
     case .none:
-      return
+      layer.mask?.removeFromSuperlayer()
```
Comment (JakeLin) [inline IBAnimatable/MaskDesignable.swift:53]: Thanks for your great demo app for picking up this issue

File: IBAnimatableApp/ParamType.swift [hunk 3]
```diff
@@ -31,14 +31,8 @@ enum ParamType {
...
   init<T: RawRepresentable>(fromEnum: T.Type) where T: Hashable {
-    // _ = iterateEnum(fromEnum)
-    //TODO: REWRITE THIS
-    /*
-    let values = iterator.map { (e: T) -> String in
-      return String(describing: e.rawValue)
-    }*/
-    //let values = iterator.map {  return String(describing: $0.rawValue) }
-    let values: [String] = []
+    let iterator = iterateEnum(fromEnum)
```
Comment (JakeLin) [inline IBAnimatableApp/ParamType.swift:34]: My bad again 😓

**Comment by JakeLin** [comment]
Great work @lastMove I have merged it

Related to #221

**Comment by lastMove** [comment]
Nice !

**Pull request merged**